src/rgw/rgw_op.cc

```diff
@@ -924,7 +924,14 @@ void handle_replication_status_header(
       bl.append("COMPLETED");
       setattrs[RGW_ATTR_OBJ_REPLICATION_STATUS] = bl;
       int ret = s->object->set_obj_attrs(dpp, &setattrs, &rmattrs, y, 0);
-      if (ret < 0) {
+      if (ret == -ECANCELED) {
+        s->object->invalidate();
+        ret = s->object->load_obj_state(dpp, y, true);
```

Comment:
are you sure we can't ignore this error and proceed with the read request? since we're racing, it seems fine to return the original "PENDING" status so we can avoid the extra rados read here